db/migrate/20151108025822_add_foreign_keyto_lists.rb

```diff
@@ -0,0 +1,11 @@
+class AddForeignKeytoLists < ActiveRecord::Migration
+  def self.up
+    add_foreign_key :book_lists, :books, on_delete: :cascade
+    add_foreign_key :book_lists, :lists, on_delete: :cascade
+  end
+
+  def self.down
+    remove_foreign_key :book_lists, :books, on_delete: :cascade
+    remove_foreign_key :book_lists, :lists, on_delete: :cascade
+  end
+end
```

Comment:
@samnang @evan-007 I just wanted to highlight how I added the foreign key's to book_lists.  Also note that I removed the dependent destroy from the models and from the spec.